subprojects/build-cache-http/src/integTest/groovy/org/gradle/caching/http/internal/HttpBuildCacheServiceErrorHandlingIntegrationTest.groovy

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.caching.http.internal
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.test.fixtures.server.http.HttpBuildCache
+import org.junit.Rule
+
+class HttpBuildCacheServiceErrorHandlingIntegrationTest extends AbstractIntegrationSpec {
+    @Rule
+    HttpBuildCache httpBuildCache = new HttpBuildCache(testDirectoryProvider)
+
+    def setup() {
+        buildFile << """   
+            import org.gradle.api.*
+            apply plugin: 'base'
+
+            @CacheableTask
+            class CustomTask extends DefaultTask {
```

Comment:
We've hit a bit of a critical mass of test `@CacheableTask` tasks... I wonder if we should pull these up into a fixture so we can just do something like `buildFile << CacheableTasks.simple()` or `buildFile << CacheableTasks.generateFile(size)`.

WDYT? (Not saying you should do it)